pkg/kubelet/kubelet_node_status.go

```diff
@@ -403,7 +403,9 @@ func (kl *Kubelet) tryUpdateNodeStatus(tryNumber int) error {
 		return fmt.Errorf("nil %q node object", kl.nodeName)
 	}
 
-	kl.updatePodCIDR(node.Spec.PodCIDR)
+	if node.Spec.PodCIDR != "" {
+		kl.updatePodCIDR(node.Spec.PodCIDR)
```

Comment:
Shall we move this check to method `updatePodCIDR`?

Reply:
Both should be ok, but keep it outside is more clear